ecllibrary/std/OpenSSL.ecl

```diff
@@ -0,0 +1,329 @@
+     * randomly-generated passphrase. The passphrase, which is much shorter,
+     * is then encrypted with the public key. The whole package is then bundled
+     * together into an "envelope" and "sealed". Given the private key that
+     * corresponds to one of the public keys used to create the ciphertext,
+     * this function unpacks everything and decrypts the payload.
+     *
+     * The function uses RSA public keys and they must be in PEM format. To
+     * generate such keys on the command line:
+     *
+     *      ssh-keygen -b 4096 -t rsa -m pem -f sample2
+     *      ssh-keygen -f sample2 -e -m pem > sample2.pub
+     *
+     * The resulting files, sample2 and sample2.pub, are the private and public
+     * keys, respectively. Their contents may be passed to this function.
+     *
+     * @param   ciphertext          The data to decrypt; REQUIRED
+     * @param   pem_private_key     An RSA public key in PEM format; REQUIRED
+     * @param   symmetric_algorithm The name of the symmetric algorithm to use
+     *                              to decrypt the payload; must be one of those
+     *                              returned by Ciphers.AvailableAlgorithms() and
+     *                              it must match the algorithm used to create the
+     *                              ciphertext; OPTIONAL, defaults to aes-256-cbc
+     *
+     * @return  The decrypted plaintext.
+     *
+     * @see     Seal()
+     *          Ciphers.AvailableAlgorithms()
+     */
+    EXPORT DATA Unseal(DATA ciphertext, STRING pem_private_key, VARSTRING symmetric_algorithm = 'aes-256-cbc') := lib_openssl.OpenSSL.rsaUnseal(ciphertext, pem_private_key, symmetric_algorithm);
+
+    /**
+     * This function performs asymmetric encryption. It should be used to
+     * encrypt only small plaintext (e.g. less than 100 bytes) because it is
+     * computationally expensive.
+     *
+     * @param   plaintext       The data to encrypt; REQUIRED
+     * @param   pem_public_key  The public key to use for encryption, in
+     *                          PEM format; REQUIRED
+     *
+     * @return  The encrypted ciphertext.
+     *
+     * @see     Decrypt()
+     */
+    EXPORT DATA Encrypt(DATA plaintext, STRING pem_public_key) := lib_openssl.OpenSSL.rsaEncrypt(plaintext, pem_public_key);
+
+    /**
+     * This function performs asymmetric decryption. It should be used to
+     * decrypt only small plaintext (e.g. less than 100 bytes) because it is
+     * computationally expensive.
+     *
+     * @param   ciphertext      The data to decrypt; REUIRED
+     * @param   pem_private_key The private key to use for decryption, in
+     *                          PEM format; REQUIRED
+     *
+     * @return  The decrypted plaintext.
+     *
+     * @see     Encrypt()
+     */
+    EXPORT DATA Decrypt(DATA ciphertext, STRING pem_private_key) := lib_openssl.OpenSSL.rsaDecrypt(ciphertext, pem_private_key);
```

Comment:
This function should permit passing a passphrase for the private key, like RSA.Sign().

Reply:
Added passphrase parameter.